packages/react-components/src/components/alarm-state/alarm-state-text.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import Box from '@cloudscape-design/components/box';
+import Icon from '@cloudscape-design/components/icon';
+import {
+  colorBorderStatusError,
+  colorBorderStatusSuccess,
+  colorBorderStatusWarning,
+  colorTextStatusInactive,
+  spaceScaledXxs,
+} from '@cloudscape-design/design-tokens';
+
+import './alarm-state-text.css';
+import { PascalCaseStateName } from '../../hooks/useAlarms/transformers';
+
+type AlarmStateTextOptions = {
+  state?: PascalCaseStateName;
+  inheritFontColor?: boolean;
+};
+
+export const AlarmStateText = ({
+  state,
+  inheritFontColor,
+}: AlarmStateTextOptions) => {
+  let icon = null;
+  let text = null;
+  let styles: React.CSSProperties = {
+    gap: spaceScaledXxs,
+    textDecoration: 'none',
+  };
+
+  let borderBottom = '1px dashed ';
+
+  switch (state) {
```

Comment:
I like how things are grouped. It's a great way to prevent bugs. I would suggest you consider how you can avoid the duplication and the mutation. I can see how someone could easily introduce a bug here by changing something on accident or forgetting to keep consistency.